std/file.d

```diff
@@ -3096,9 +3096,8 @@ else version(Posix)
         private this(string path, core.sys.posix.dirent.dirent* fd) @trusted
         {
             import std.path : buildPath;
-            import std.algorithm.searching : countUntil;
 
-            immutable len = fd.d_name[].countUntil(0);
```

Comment:
I'm not clear on why calling `countUntil` truncates the file name. Is it somehow due to autodecoding?

Reply:
Yes. `countUntil` uses ranges, so it auto-decodes and counts code points, whereas `strlen` just counts the number of chars until it hits the null character.